### PR TITLE
chore(dependencies): Upgrade Spring Boot to 2.2.1

### PR DESCRIPTION
**Pull request by scottfrederick**
See https://github.com/spinnaker/spinnaker/issues/5134

Dependent on https://github.com/spinnaker/kork/pull/419 for a Spring Boot dependency upgrade

### CODE REVIEW

**Comment by cfieber** [comment]
this one troubles me a bit with the typeref changes requiring more specialized types on the left hand side - going to poke around a bit and see if there are any other options there (and live with it as is if we have to)

File: cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cache/AbstractRedisCache.java [hunk 1]
```diff
@@ -234,7 +234,7 @@ protected String allOfTypeId(String type) {
     return String.join(":", prefix, type, "members");
   }
 
-  protected TypeReference getRelationshipsTypeReference() {
+  protected TypeReference<?> getRelationshipsTypeReference() {
```
Comment (cfieber) [inline cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cache/AbstractRedisCache.java]:
```suggestion
  protected TypeReference<? extends Collection<String>> getRelationshipsTypeReference() {
```

File: cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cache/RedisCache.java [hunk 2]
```diff
@@ -306,7 +306,7 @@ private CacheData extractItem(String id, List<String> keyResult, List<String> kn
         if (rel != null) {
           String relType = knownRels.get(relIdx - 1);
           Collection<String> deserializedRel =
-              objectMapper.readValue(rel, getRelationshipsTypeReference());
+              (Collection<String>) objectMapper.readValue(rel, getRelationshipsTypeReference());
```
Comment (cfieber) [inline cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cache/RedisCache.java]: cast should be unnecessary with the above comment

File: cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cache/AbstractRedisCache.java (hunk 1 above)
Comment (scottfrederick) [reply cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cache/AbstractRedisCache.java]: Good catch. Any improvement in the types is a good thing. Resolved in 43f237c16f37208c069aceec126271f773eb57b9

**Pull request merged**